pipelines/riff.yml

```diff
@@ -460,6 +476,89 @@ jobs: ##########################################################################
 
 ## Jobs: Components ##
 
+- name: build-component-images
+  plan:
+  - aggregate: # fetch resources
+    - get: git-riff
+      params:
+        depth: 1
+    - get: git-pfs-ci
+    - get: riff-version
+
+  - aggregate: # pre-build metadata
+    - put: riff-version
+      params:
+        pre: build
+    - put: github-commit-status-riff
+      params:
+        commit: git-riff
+        state: pending
+        description: Building, testing and creating docker images for riff server components
+
+  - aggregate: # build and test
+    - task: test-topic-controller
+      file: git-pfs-ci/tasks/riff-components/test/task.yml
```

Comment:
I thought we agreed on not trying to factorize behavior across components.
I believe this PR will work, but then we should at least clean unused files

Reply:
Only sidecar needs the kafka container, all the others work with the `test` task. I did notice that while I updated that task (`test-function-sidecar`) in the `riff` pipeline, it still have a `params` block hanging off from when I tried to do it using the factorized `test` task.

I had in mind to push the cleanup into a following commit. Not that I don't want to do it, but I'd like to get to the working case first.